### PR TITLE
Fix Spectrogram problem because of webkit's OfflineAudioContext

### PR DESCRIPTION
**Pull request by fzyukio**
Use my own FFT calculater instead of relying on the broken webkit. Now all browsers are supported and the calculation is much much faster!.

To use:

`     ws.Spectrogram.init ( (

      wavesurfer: self.ws,
      fftSamples: 512,
      noverlap: 256, // But it's better to leave this param unset - the program will choose the best value based on canvas width to maximise resolution.
      pixelRatio: 1,
      windowFunc: 'hann',
      container: "#waveform-spectrogram"
});
`

Provide 10 different window functions to choose from.
FFT calculator is based on @corbanbrook library (https://github.com/corbanbrook/dsp.js)

### CODE REVIEW

File: plugin/wavesurfer.spectrogram.js [hunk 1]
```diff
@@ -230,7 +247,184 @@ WaveSurfer.Spectrogram = {
         }
 
         return newMatrix;
-    }
+    },
```
Comment (katspaugh) [inline plugin/wavesurfer.spectrogram.js]: A trailing comma (not critical).

File: plugin/wavesurfer.spectrogram.js [hunk 2]
```diff
@@ -230,7 +247,184 @@ WaveSurfer.Spectrogram = {
...
+
 };
 
 WaveSurfer.util.extend(WaveSurfer.Spectrogram, WaveSurfer.Observer);
+
+function FFT(bufferSize, sampleRate, windowFunc, alpha) {
```
Comment (katspaugh) [inline plugin/wavesurfer.spectrogram.js]: Maybe store this function in `WaveSurfer.FFT`?

Comment (katspaugh) [inline plugin/wavesurfer.spectrogram.js]: Could you also add a comment about the code being based on https://github.com/corbanbrook/dsp.js?

**Comment by katspaugh** [comment]
すごいです。:bow:

screen shot 2016-02-19 at 10 36 50
screen shot 2016-02-19 at 10 37 53

File: plugin/wavesurfer.spectrogram.js (hunk 2 above)
Comment (fzyukio) [inline plugin/wavesurfer.spectrogram.js]: ( ͡° ͜ʖ ͡°)

**Comment by fzyukio** [comment]
@katspaugh ありがとう　:+1:

**Comment by katspaugh** [comment]
@akreal I hope you don't mind :smile:

**Comment by akreal** [comment]
@katspaugh Sure!

**Pull request merged**